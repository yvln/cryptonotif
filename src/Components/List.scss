.List {
  display: flex;
  justify-content: center;
  flex-direction: column;

  @media screen and (min-width: 480px) {
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
  }
}
