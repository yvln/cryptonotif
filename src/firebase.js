/* eslint-disable */

const firebase = require('firebase');
require('dotenv').config();

const config = {
  apiKey: process.env.API_KEY,
  authDomain: 'cryptonotif-56a23.firebaseapp.com',
  databaseURL: 'https://cryptonotif-56a23.firebaseio.com',
  projectId: 'cryptonotif-56a23',
  storageBucket: 'cryptonotif-56a23.appspot.com',
  messagingSenderId: '1008924569797',
};

export const initializeFirebase = () => {
  firebase.initializeApp(config);
  return true;
};

export const getDatabase = () => {
  return firebase.database();
};
