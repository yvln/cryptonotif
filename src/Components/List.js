import React, { Component } from "react";

import { getDatabase } from "../firebase";
import Alert from "./Alert";
import "./List.scss";

class List extends Component {
  state = {
    alerts: []
  };

  componentDidMount() {
    getDatabase()
      .ref("/alert")
      .on("value", snapshot => {
        const values = snapshot.val();

        if (!values) {
          return;
        }

        return this.setState({
          alerts: Object.keys(values).map(value => values[value])
        });
      });
  }

  renderAlert = (alert, i) => <Alert id={i} key={alert.key} data={alert} />;

  renderAlertsList = () => {
    return this.state.alerts.map(this.renderAlert);
  };

  render() {
    return <div className="List">{this.renderAlertsList()}</div>;
  }
}

export default List;
