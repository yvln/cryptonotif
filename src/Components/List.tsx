import React, { Component } from "react";

import { getDatabase } from "../firebase";
import Alert from "./Alert";
import "./List.scss";

type Data = {
  action: string;
  amount: string;
  asset: string;
  currency: string;
  email: string;
  key: string;
}

type ListState = {
  alerts: Data[];
}

class List extends Component<void, ListState> {
  state = {
    alerts: []
  };

  componentDidMount() {
    getDatabase()
      .ref("/alert")
      .on("value", snapshot => {
        if (!snapshot) {
          return;
        }

        const values = snapshot.val();

        if (!values) {
          return;
        }

        return this.setState({
          alerts: Object.values(values)
        });
      });
  }

  renderAlert = (alert: Data, i: number) => <Alert id={i} key={alert.key} data={alert} />;

  renderAlertsList = () => {
    return this.state.alerts.map(this.renderAlert);
  };

  render() {
    return <div className="List">{this.renderAlertsList()}</div>;
  }
}

export default List;
