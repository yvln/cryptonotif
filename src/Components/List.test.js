import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import renderer from "react-test-renderer";

import { initializeFirebase } from "../firebase";

import List from "./List";

configure({ adapter: new Adapter() });

describe("src/Components/List", () => {
  it("should render", () => {
    initializeFirebase();
    const component = renderer.create(<List />);

    expect(component).toMatchSnapshot();
  });
});
