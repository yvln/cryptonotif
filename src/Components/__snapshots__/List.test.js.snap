// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/Components/List should render 1`] = `
<div
  className="List"
/>
`;
